beacon_node/network/src/sync/network_context.rs

```diff
@@ -473,14 +487,18 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         &self,
         request: BlocksByRangeRequest,
         custody_indexes: &HashSet<ColumnIndex>,
+        syncing_peers: impl Iterator<Item = PeerId>,
     ) -> Result<HashMap<PeerId, DataColumnsByRangeRequest>, RpcRequestSendError> {
         let mut peer_id_to_request_map = HashMap::new();
+        let syncing_peers = syncing_peers.collect::<Vec<_>>();
 
         for column_index in custody_indexes {
             // TODO(das): The peer selection logic here needs to be improved - we should probably
             // avoid retrying from failed peers, however `BatchState` currently only tracks the peer
             // serving the blocks.
-            let Some(custody_peer) = self.get_random_custodial_peer(*column_index) else {
+            let Some(custody_peer) =
+                self.choose_random_custodial_peer(*column_index, syncing_peers.iter())
```

Comment:
I think this change should not be bundled into this PR. Range sync syncing chains are not filled with peers immediately, instead, there's some delay as peers are status'd. I worry that
- Syncing is created
- In our global peer set we have enough peers on each column, but in this syncing chain there's only few peers
- Syncing chain attempts to make progress and passes the small set of peers to this function
- `Err(RpcRequestSendError::NoCustodyPeers)` condition is hit rather frequently

To do this change properly I think the syncing chains must be able to be "peer-less" for some time. Similar to lookups, but that would take some refactoring.

The fact that range sync draws from the global set of peers is a known design tradeoff, unrelated to making PeerDAS Fulu compatible